### ISSUE TITLE
Add edge labels

### ISSUE THREAD

**Issue by JackDevine** [JuliaPlots/GraphRecipes.jl#58]
cc: @keorn

An example of another library with this feature is TikzGraphs:
```julia
using TikzGraphs
using LightGraphs
using TikzPictures

g = DiGraph(4)
add_edge!(g, 2, 1)
add_edge!(g, 2, 3)
add_edge!(g, 3, 4)
add_edge!(g, 1, 4)

plot(g, ["A", "B", "C", "D"], edge_labels=Dict((2,1) => L"x", (1,4) => "four to one"))
```
[edge_labels.pdf](https://github.com/JuliaPlots/GraphRecipes.jl/files/3259275/edge_labels.pdf)

I think that their API is good and we should just copy that. I am willing to do the work on this as long as people agree that using a dictionary for the edge labels is a good idea.

I am also willing to bikeshed the name `edge_labels`.

**Comment by mkborregaard** [comment]
yeah great. It couldn't use the existing Plots API, as `series_annotations` codes for node labels. I think allowing the user to pass either a Dict or Vector ordered as the edge matrix would be great, to offer flexibility. The same API could be used for edge labels on phylogenies (https://github.com/richardreeve/Phylo.jl/pull/15 ) or cluster dendrograms ( https://github.com/JuliaPlots/StatsPlots.jl#dendrograms ) so I don't know - might even be added up into Plots?
WRT the name, `edge_labels` and `edgelabels` could be synonymous keywords IMHO.

**Comment by JackDevine** [comment]
I didn't think of the vector API, good catch! Would we use `missing` for all of the edges that don't have labels?

Directed graphs have a meaningful order for a Vector of labels. Can we be sure that any other data type that has edges will also come with a sensible order for those edges? It seems to me that there has to be some underlying matrix representation for that to be true. I will prototype `edgelabels` for Plots.jl and return with my findings.

**Comment by mkborregaard** [comment]
I think we can assume that there is an order, yes, though for dendrograms etc there are several alternatives, depthfirst, breadthfirst etc. I think we can assume the user to pass in their labels in the same order as the edges, or else use the Dict possibility.
I think we could be lenient and allow `missing`, `NaN`, `""` and `nothing` to all mean no label.

**Comment by JackDevine** [comment]
Thank you very much for your input!

I will get onto this as soon as I can. I will straighten out a few other issues first, so if others want to express their thoughts, then there is still some time.

**Comment by JackDevine** [comment]
This was closed by #65
